src/main/java/io/kestra/plugin/kestra/ee/tests/RunTests.java

```diff
@@ -0,0 +1,175 @@
+@Schema(
+    title = "Run multiple unit tests",
+    description = "Run unit tests at multiple levels: all tests for a tenant or namespace, all tests for a specific flow, or a test suite by ID."
+)
+@Plugin(
+    examples = {
+        @Example(
+            title = "Run all current tenant tests",
+            full = true,
+            code = """
+                id: run_tenant_tests
+                namespace: company.team
+
+                tasks:
+                  - id: run_tenant_all_tests
+                    type: io.kestra.plugin.kestra.ee.tests.RunTests
+                    auth:
+                      apiToken: "{{ secret('KESTRA_API_TOKEN') }}"
+                """
+        )
+    }
+)
+public class RunTests extends AbstractKestraTask implements RunnableTask<RunTests.Output> {
+    @Schema(title = "The namespace")
+    @Nullable
+    private Property<String> namespace;
+
+    @Schema(title = "To include child namespaces or not")
+    @Builder.Default
+    private Property<Boolean> includeChildNamespaces = Property.ofValue(true);
+
+    @Schema(title = "The Flow id")
+    @Nullable
+    private Property<String> flowId;
+
+    @Schema(title = "Should the task be marked as FAILED when a test fails")
+    @Builder.Default
+    private Property<Boolean> failOnTestFailure = Property.ofValue(false);
+
+    @Override
+    public Output run(RunContext runContext) throws Exception {
+        var kestraClient = kestraClient(runContext);
+        var testSuitesApi = kestraClient.testSuites();
+
+        var rTenantId = runContext.render(tenantId).as(String.class).orElse(runContext.flowInfo().tenantId());
+        var rNamespace = runContext.render(namespace).as(String.class).orElse(null);
+        var rIncludeChildNamespaces = runContext.render(includeChildNamespaces).as(Boolean.class).orElse(true);
+        var rFlowId = runContext.render(flowId).as(String.class).orElse(null);
+        var rFailOnTestFailure = runContext.render(failOnTestFailure).as(Boolean.class).orElse(false);
+
+        var runByQueryRequest = new TestSuiteServiceRunByQueryRequest()
+            .namespace(rNamespace)
+            .includeChildNamespaces(rIncludeChildNamespaces)
+            .flowId(rFlowId);
+        runContext.logger().info("Running tests for query: namespace: '{}', includeChildNamespaces: '{}', flowId: '{}'", runByQueryRequest.getNamespace(), runByQueryRequest.getIncludeChildNamespaces(), runByQueryRequest.getFlowId());
+
+        var result = testSuitesApi.runTestSuitesByQuery(rTenantId, runByQueryRequest);
+        Objects.requireNonNull(result.getResults());
+        runContext.logger().info("Requested to run {} test suites, {} test cases", result.getNumberOfTestSuitesToBeRun(), result.getNumberOfTestCasesToBeRun());
+
+        var outputBuilder = Output.builder().result(result);
+        AtomicReference<Optional<State.Type>> errorState = new AtomicReference<>(Optional.empty());
+        result.getResults().forEach(testSuiteRunResult -> {
```

Comment:
Hmm I feel like you could just avoid duplication here and use `RunTest` from `RunTests` ?

Reply:
I dont think so, the logic is slightly different